utils/grpc/setup_server.go

```diff
@@ -0,0 +1,121 @@
+package grpc
+
+import (
+	"log"
+	"net"
+
+	"github.com/tron-us/go-btfs-common/protos/escrow"
+	"github.com/tron-us/go-btfs-common/protos/guard"
+	"github.com/tron-us/go-btfs-common/protos/hub"
+	"github.com/tron-us/go-btfs-common/protos/shared"
+	"github.com/tron-us/go-btfs-common/protos/status"
+
+	"github.com/tron-us/go-common/v2/constant"
+	"github.com/tron-us/go-common/v2/middleware"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/reflection"
+)
+
+type GrpcServer struct {
+	server       *grpc.Server
+	healthServer *health.Server
+	serverName   string
+	lis          net.Listener
+	dBURL        string
+	rDURL        string
+}
+
+func (s *GrpcServer) serverTypeToServerName(server interface{}) {
+	switch server.(type) {
+	case status.StatusServiceServer:
+		s.serverName = "status-server"
+	case escrow.EscrowServiceServer:
+		s.serverName = "escrow"
+	case guard.GuardServiceServer:
+		s.serverName = "guard"
+	case hub.HubQueryServiceServer:
+		s.serverName = "hub"
+	default:
+		s.serverName = "status-server"
```

Comment:
please default to `unknown`, this is for a new implementor to notice this and add accordingly

Reply:
ok.